src/pmdas/uwsgi/pmdauwsgi.python

```diff
@@ -0,0 +1,111 @@
+        self.add_metric(name + '.workers_accepting_requests', pmdaMetric(PMDA.pmid(0, 3), PM_TYPE_U64, PM_INDOM_NULL, PM_SEM_COUNTER, pmUnits()))
+        self.add_metric(name + '.total_exceptions', pmdaMetric(PMDA.pmid(0, 4), PM_TYPE_U64, PM_INDOM_NULL, PM_SEM_COUNTER, pmUnits()))
+        self.add_metric(name + '.total_harakiri_count', pmdaMetric(PMDA.pmid(0, 5), PM_TYPE_U64, PM_INDOM_NULL, PM_SEM_COUNTER, pmUnits()))
+        self.add_metric(name + '.busy_worker_count', pmdaMetric(PMDA.pmid(0, 6), PM_TYPE_U64, PM_INDOM_NULL, PM_SEM_COUNTER, pmUnits()))
+        self.add_metric(name + '.idle_worker_count', pmdaMetric(PMDA.pmid(0, 7), PM_TYPE_U64, PM_INDOM_NULL, PM_SEM_COUNTER, pmUnits()))
+        self.add_metric(name + '.pause_worker_count', pmdaMetric(PMDA.pmid(0, 8), PM_TYPE_U64, PM_INDOM_NULL, PM_SEM_COUNTER, pmUnits()))
+
+        self.set_fetch_callback(self.uwsgi_fetch_callback)
+        self.set_user(pmContext.pmGetConfig('PCP_USER'))
+
+    def uwsgi_fetch_callback(self, cluster, item, inst):
+        ''' Returns a list of value,status (single pair) for one metric '''
+
+        if not cluster == 0:
+            return [PM_ERR_PMID, 0]
+        elif not _is_pmda_setup():
+            try:
+                stats = requests.get('http://127.0.0.1:9051').json()
+                self.log(f"stats: {stats}")
+            except Exception:
+                return [PM_ERR_AGAIN, 0]
+
+        if item == 0:
+            return [len(stats['workers']), 1]
+        if item == 1:
+            response_time_msec_per_worker = [w['avg_rt'] for w in stats['workers']] # reports in milliseconds
+            avg_response_time_msec = sum(response_time_msec_per_worker)/len(response_time_msec_per_worker)
+            return [avg_response_time_msec, 1]
+        if item == 2:
+            total_requests_served = sum([w['requests'] for w in stats['workers']])
+            return [total_requests_served, 1]
+        if item == 3:
+            total_accepting_requests = sum([w['accepting'] for w in stats['workers']])
+            return [total_accepting_requests, 1]
+        if item == 4:
+            total_exceptions = sum([w['exceptions'] for w in stats['workers']])
+            return [total_exceptions, 1]
+        if item == 5:
+            total_harakiri_count = sum([w['harakiri_count'] for w in stats['workers']])
+            return [total_harakiri_count, 1]
+        if item == 6:
+            num_busy_workers = sum([w['status'] == 'busy' for w in stats['workers']])
+            return [num_busy_workers, 1]
+        if item == 7:
+            num_idle_workers = sum([w['status'] == 'idle' for w in stats['workers']])
+            return [num_idle_workers, 1]
+        if item == 8:
+            num_pause_workers = sum([w['status'] == 'pause' for w in stats['workers']])
+            return [num_pause_workers, 1]
+        return [PM_ERR_PMID, 0]
```

Comment:
I notice all metrics are aggregates across the 'workers' domain - should we have separate metrics for totals vs individual workers?  (i.e. an instance domain, in PCP terminology).  See pmdalio.python for some example code.

Reply:
@natoscott in case of separating the totals vs individual workers, does the summary is also consider as one of the instance domain ? Because when I do that I am getting following issue:

```
[Wed Feb  7 11:07:02] pmdauwsgi(404761) Error: prefetch: callback failed
Traceback (most recent call last):
  File "/var/lib/pcp/pmdas/uwsgi/pmdauwsgi.python", line 252, in uwsgi_fetch
  File "/usr/lib64/python3.9/site-packages/pcp/pmda.py", line 375, in replace_indom
    replacement = pmdaIndom(it_indom, insts)
  File "/usr/lib64/python3.9/site-packages/pcp/pmda.py", line 142, in __init__
    self.set_instances(indom, insts)
  File "/usr/lib64/python3.9/site-packages/pcp/pmda.py", line 210, in set_instances
    self.it_numinst = len(insts) # signifies an old-school array indom
TypeError: object of type 'UWSGISUMMARY' has no len()
```

Reply:
nvm, I think I got it. It doesn't nned to be an instance domain.

Reply:
Done